proto/client.proto

```diff
@@ -51,7 +51,14 @@ message UpdateRequest {
 	Profile profile = 2	[(gogoproto.customtype) = "EncodedProfile", (gogoproto.nullable) = false];
 	LookupRequest lookup_parameters = 3;
 
-	bytes dkim_proof = 1000 [(gogoproto.customname) = "DKIMProof"];
+	oneof email_proof {
+		bytes dkim_proof = 1000 [(gogoproto.customname) = "DKIMProof"];
+		// oidc_token contains OpenID Connect token received from the provider
+		string oidc_token = 1001 [(gogoproto.customname) = "OIDCToken"];
+		// auth_data is an opaque field and is primarily provided in case keyserver
+		// wants to do custom verification to confirm ownership of an email address
+		bytes auth_data = 1002;
```

Comment:
Another option to consider is to separate `email_proof` into its own message type as is done for the corresponding keyserver configuration option below. Then we wouldn't have to set aside a range of values manually. But that would break wire-format compatibility (which we can do now, I guess).